apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -211,3 +261,232 @@ export async function validateLti13CourseInstance(
     z.boolean(),
   );
 }
+
+export async function getAccessToken(lti13_instance_id: string) {
+  const lti13_instance = await selectLti13Instance(lti13_instance_id);
+
+  let tokenSet: TokenSet = lti13_instance.access_tokenset;
+
+  // -5 minute buffer to refresh tokens before they expire
+  if (
+    lti13_instance.access_token_expires_at &&
+    lti13_instance.access_token_expires_at > new Date(Date.now() - 5 * 60 * 1000)
+  ) {
+    return tokenSet.access_token;
+  } else {
```

Comment:
Nit: with the early return above, you can drop the `else` to reduce indentation.

Reply:
Addressed in 46c87e028